src/main/java/org/openstreetmap/atlas/utilities/command/abstractcommand/AbstractAtlasShellToolsCommand.java

```diff
@@ -684,17 +678,36 @@ protected void runSubcommandAndExit(final String... args)
             {
                 printlnStdout(this.getHelpMenu());
             }
-            System.exit(0);
+            return 0;
         }
 
         if (this.parser.hasOption(VERSION_OPTION_LONG))
         {
             printlnStdout(String.format("%s version %s", getCommandName(), this.version));
-            System.exit(0);
+            return 0;
         }
 
         // run the command
-        System.exit(execute());
+        return execute();
+    }
+
+    /**
+     * Run this subcommand and exit the JVM. An example of how this method should be called to make
+     * the command functional with an external wrapper:
+     *
+     * <pre>
+     * public static void main(final String[] args)
+     * {
+     *     new MySubclassSubcommand().runSubcommandAndExit(args);
+     * }
+     * </pre>
+     *
+     * @param args
+     *            the command arguments
+     */
+    public void runSubcommandAndExit(final String... args)
```

Comment:
This was `protected` before. Is there a reason you're switching to `public`?

Reply:
I needed runSubcommand to be public so that it could be accessed for unit tests. I did not think it made a lot of sense to have runSubcommandAndExit remain protected when it is basically just calling the public runSubcommand.